### PR TITLE
Refactor: use NewClient() interface to create a scm client

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by 6543** [comment]
@gabrie30 ping - this refactor makes scm package more independent from the rest of the code

File: scm/gitea.go [hunk 1]
```diff
@@ -18,7 +18,12 @@ func init() {
 	registerClient(Gitea{})
 }
 
-type Gitea struct{}
+type Gitea struct {
+	// contain the gitea client
```
Comment (6543) [inline scm/gitea.go:22]:
```suggestion
	// extend the gitea client
```

File: scm/github.go [hunk 2]
```diff
@@ -20,7 +20,10 @@ func init() {
 }
 
 type Github struct {
-	client *github.Client
+	// contain the github client
```
Comment (6543) [inline scm/github.go:23]:
```suggestion
	// extend the github client
```

**Comment by gabrie30** [comment]
awesome thanks @6543 !

**Comment by 6543** [comment]
@gabrie30 was #106 the only reason to revert #105 or do I have missed something here?

**Comment by gabrie30** [comment]
@6543 you raised https://github.com/gabrie30/ghorg/pull/105 and https://github.com/gabrie30/ghorg/pull/106 I merged 105 then went to merge 106 and there were merge conflicts so i reverted 105, that revert was https://github.com/gabrie30/ghorg/pull/107, so currently none of your changes are on master

**Pull request merged**